tests/test_io/test_surface_reader.py

```diff
@@ -25,13 +37,19 @@ def test_dataframes(get_unstructured_data):
 
 def test_edges_shape(get_unstructured_data):
     assert get_unstructured_data.edges.shape[1] == 4
-#def test_plot_pyvista(get_unstructured_data):
-    # tm = TetraMesh(get_unstructured_data)
-    # s = to_pyvista_tetra(tm)
-    # pv_plot([s], image_2d=True)
-    # topo_df = pd.read_csv(input_path + "/land_surface_vertices.csv")
-    # topo = pv.PolyData(topo_df.values).delaunay_2d()
-    # topo.plot(notebook=False, show_grid=True)
+
+
+def test_unstructured_element(get_less_unstructured_data):
+    tm = TetraMesh(get_less_unstructured_data)
+    assert len(tm.tetrahedrals) == 39
+    assert isinstance(tm, TetraMesh)
+
+
+def test_plot_pyvista(get_less_unstructured_data):
+    tm = TetraMesh(get_less_unstructured_data)
+    s = to_pyvista_tetra(tm) # Process finished here with exit code 139 (interrupted by signal 11: SIGSEGV)
+    assert isinstance(s, UnstructuredGrid)
+    pv_plot([s], image_2d=True)
```

Comment:
if I comment line 52 the test is passed, but I get this message:

Testing started at 15:30 ...
/home/monique/[/opt/anaconda3]/envs/gempy_env_py3.8/bin/python /opt/pycharm-professional-2020.2.2/pycharm-2020.2.2/plugins/python/helpers/pycharm/_jb_pytest_runner.py --target test_surface_reader.py::test_plot_pyvista
Launching pytest with arguments test_surface_reader.py::test_plot_pyvista in /home/monique/Documents/subsurface/tests/test_io

============================= test session starts ==============================
platform linux -- Python 3.8.5, pytest-6.0.1, py-1.9.0, pluggy-0.13.1 -- /home/monique/[/opt/anaconda3]/envs/gempy_env_py3.8/bin/python
cachedir: .pytest_cache
rootdir: /home/monique/Documents/subsurface
collecting ... collected 1 item

test_surface_reader.py::test_plot_pyvista 

=============================== warnings summary ===============================
/home/monique/[/opt/anaconda3]/envs/gempy_env_py3.8/lib/python3.8/site-packages/pyvista/core/pyvista_ndarray.py:2
  /home/monique/[/opt/anaconda3]/envs/gempy_env_py3.8/lib/python3.8/site-packages/pyvista/core/pyvista_ndarray.py:2: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3, and in 3.9 it will stop working
    from collections import Iterable

tests/test_io/test_surface_reader.py::test_plot_pyvista
tests/test_io/test_surface_reader.py::test_plot_pyvista
  /home/monique/[/opt/anaconda3]/envs/gempy_env_py3.8/lib/python3.8/site-packages/vtkmodules/util/numpy_support.py:66: DeprecationWarning: Converting `np.character` to a dtype is deprecated. The current result is `np.dtype(np.str_)` which is not strictly correct. Note that `np.character` is generally deprecated and 'S1' should be used.
    if numpy_array_type == key or \

tests/test_io/test_surface_reader.py::test_plot_pyvista
tests/test_io/test_surface_reader.py::test_plot_pyvista
  /home/monique/[/opt/anaconda3]/envs/gempy_env_py3.8/lib/python3.8/site-packages/vtkmodules/util/numpy_support.py:68: DeprecationWarning: Converting `np.character` to a dtype is deprecated. The current result is `np.dtype(np.str_)` which is not strictly correct. Note that `np.character` is generally deprecated and 'S1' should be used.
    numpy_array_type == numpy.dtype(key):

-- Docs: https://docs.pytest.org/en/stable/warnings.html
======================== 1 passed, 5 warnings in 0.49s =========================

Process finished with exit code 0
PASSED                         [100%]


Have you encountered a similar problem yet? If so I would appreciate your help, otherwise I'll google my way to a solution

Reply:
The problem is that Delauny is returning 4 edges and therefor a 3d volume instead a 2D surface in 3D space: https://stackoverflow.com/questions/26434726/return-surface-triangle-of-3d-scipy-spatial-delaunay

- Assuming that the layers are horizontal an easy way to solve it would be to use only the XY coord of the vertex:
`edges = Delaunay(vertex[:, :2]).simplices` but this would not solve the problem for layers that are sub-vertical

I am still looking

Reply:
Okay using vtk for Delaunay solves the problem. Also the element is a TriSurf not a Tet since we are looking a 2D surfaces.

Now it runs you can just pull my change :)

Reply:
Thanks a lot!